Library/Homebrew/diagnostic.rb

```diff
@@ -755,6 +755,25 @@ def check_user_path_3
         end
       end
 
+
+      def check_xdg_data_dirs
+        share = "#{HOMEBREW_PREFIX}/share"
+        homebrew_in_xdg_data_dirs = true
+        if ENV.key?("XDG_DATA_DIRS")
+          xdg_data_dirs = ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR)
+          xdg_data_dirs.length > 0 && homebrew_in_xdg_data_dirs = xdg_data_dirs.include?(share)
+        end
+        unless homebrew_in_xdg_data_dirs
+          <<-EOS.undent
+          Homebrew's share was not found in your XDG_DATA_DIRS but you have
+          this variable set to include other locations.
+          Some programs like `vapigen` may not work correctly.
+          Consider setting the XDG_DATA_DIRS for example like so
+              echo 'export XDG_DATA_DIRS="#{share}:$XDG_DATA_DIRS"' >> #{shell_profile}
+        EOS
+        end
```

Comment:
Not so much ;)

Tried to do it in a more compact way, but I think logic won and I lost...